packages/arb-token-bridge-ui/src/components/Sidebar/SidebarMenu.tsx

```diff
@@ -0,0 +1,387 @@
+  link?: string
+}
+
+const stringToKey = (str: string) => str.split(' ').join('_').toLowerCase()
+
+const MenuItem = ({
+  menu,
+  activeMenu,
+  children,
+  className
+}: {
+  menu: MenuConfig
+  activeMenu: string
+  children?: React.ReactNode
+  className?: string
+}) => {
+  const { sidebarOpened } = useSidebarStore()
+
+  const menuClasses = twMerge(
+    'group flex items-center md:rounded px-[12px] py-2 text-base font-light hover:bg-default-black-hover cursor-pointer hover:opacity-100 hover:text-white',
+    sidebarOpened ? 'gap-x-3 min-w-[200px]' : 'flex-col px-[8px]',
+    activeMenu === menu.id && 'text-white bg-white/20',
+    className
+  )
+
+  if (menu.isExternalLink) {
+    return (
+      <ExternalLink
+        href={menu.link}
+        className={menuClasses}
+        onClick={menu.onClick}
+      >
+        {children}
+      </ExternalLink>
+    )
+  }
+
+  if (menu.link) {
+    return (
+      <Link href={menu.link} className={menuClasses} onClick={menu.onClick}>
+        {children}
+      </Link>
+    )
+  }
+
+  return (
+    <button className={menuClasses} onClick={menu.onClick}>
+      {children}
+    </button>
+  )
+}
+
+const SubMenuItem = ({
+  onClick,
+  children,
+  isActive,
+  link,
+  isExternalLink = false
+}: {
+  onClick?: () => void
+  children?: React.ReactNode
+  isActive?: boolean
+  link?: string
+  isExternalLink?: boolean
+}) => {
+  const subMenuClasses = twMerge(
+    'group ml-8 flex min-w-[175px] cursor-pointer items-center justify-between md:rounded py-1 pl-4 text-white/60 hover:bg-default-black-hover hover:text-white',
+    isActive && 'text-white'
+  )
+
+  if (isExternalLink) {
+    return (
+      <ExternalLink href={link} className={subMenuClasses} onClick={onClick}>
+        {children}
+
+        <ArrowTopRightOnSquareIcon className="mr-3 h-4 w-4 opacity-50 group-hover:opacity-100" />
+      </ExternalLink>
+    )
+  }
+
+  if (!isExternalLink && link) {
+    return (
+      <Link href={link} className={subMenuClasses} onClick={onClick}>
+        {children}
+      </Link>
+    )
+  }
+
+  return (
+    <button className={subMenuClasses} onClick={onClick}>
+      {children}
+    </button>
+  )
+}
+
+export const SidebarMenu = ({
+  menuItemClickCallback,
+  className
+}: {
+  menuItemClickCallback?: () => void
+  className?: string
+}) => {
+  const posthog = usePostHog()
+  const { sidebarOpened, activeMenu, setActiveMenu } = useSidebarStore()
+
+  function sendClickEventForGroup(groupTitle: string) {
+    posthog?.capture('Sidebar Menu Dropdown Click', {
+      Group: groupTitle
+    })
+  }
+
+  function sendClickEventForLink(linkTitle: string) {
+    posthog?.capture('Sidebar Menu Link Click', {
+      Link: linkTitle
+    })
+  }
+
+  function sendClickEventForMenuItem(menuItem: string) {
+    posthog?.capture('Sidebar Menu Item Click', {
+      Item: menuItem
+    })
+  }
+
+  const toggleActiveMenu = (menuId: string) => {
+    if (activeMenu === menuId) {
+      setActiveMenu('')
```

Comment:
I don't think that's how it should work. We should do nothing if the user clicks the same menu, otherwise it toggles on and off.

Reply:
Mentioned it before but there's another issue with this: query params get reset. That means that amount and selected networks get reset on bridge click. It shouldn't be the case